arcgis-ios-sdk-samples/Layers/Identify KML features/IdentifyKMLFeaturesViewController.swift

```diff
@@ -66,9 +66,14 @@ extension IdentifyKMLFeaturesViewController: AGSGeoViewTouchDelegate {
         geoView.callout.dismiss()
         geoView.identifyLayer(forecastLayer, screenPoint: screenPoint, tolerance: 15, returnPopupsOnly: false) { [weak self] (result) in
             if let error = result.error {
-                print("Error identifying layer: \(error)")
-            } else if let placemarkIndex = result.geoElements.firstIndex(where: { $0 is AGSKMLPlacemark }) {
-                let placemark = result.geoElements[placemarkIndex] as! AGSKMLPlacemark
+                self?.presentAlert(error: error)
+            } else if let placemark = result.geoElements.first(where: { $0 is AGSKMLPlacemark }) as? AGSKMLPlacemark {
```

Comment:
If `first(where:)` returns a non-nil value, the second cast is always going to succeed, which is why it was originally written the way it was.

Reply:
In this case, I'll change to `if let placemark = result.geoElements.first as? AGSKMLPlacemark { ... }`, since the `AGSGeoView.identifyLayer(_:screenPoint:tolerance:returnPopupsOnly:completion:)` method only returns the top-most identified feature in the layer. Directly checking the first should be sufficient.

![topmost](https://user-images.githubusercontent.com/9660181/111357130-4184ff80-8646-11eb-8d7f-357135515270.png)